tests/test_hvac_csc.py

```diff
@@ -0,0 +1,567 @@
+
+            tel_vex03P04 = await self.assert_next_sample(topic=self.remote.tel_vex03P04, flush=True,)
+            self.assertTrue(tel_vex03P04.fallaTermica is False)
+            self.assertTrue(tel_vex03P04.comandoEncendido is False)
+
+            comando_encendido = True
+            await self.remote.cmd_vex03P04.set_start(
+                comandoEncendido=comando_encendido, timeout=STD_TIMEOUT,
+            )
+            tel_vex03P04 = await self.assert_next_sample(topic=self.remote.tel_vex03P04, flush=True,)
+            tel_vex03P04 = await self.assert_next_sample(topic=self.remote.tel_vex03P04, flush=True,)
+            self.assertTrue(tel_vex03P04.fallaTermica is False)
+            self.assertTrue(tel_vex03P04.comandoEncendido == comando_encendido)
+
+    async def test_do_vex04P04(self):
+        async with self.make_csc(initial_state=salobj.State.STANDBY, config_dir=None, simulation_mode=1):
+            await salobj.set_summary_state(remote=self.remote, state=salobj.State.ENABLED)
+
+            tel_vex04P04 = await self.assert_next_sample(topic=self.remote.tel_vex04P04, flush=True,)
+            self.assertTrue(tel_vex04P04.fallaTermica is False)
+            self.assertTrue(tel_vex04P04.comandoEncendido is False)
+
+            comando_encendido = True
+            await self.remote.cmd_vex04P04.set_start(
+                comandoEncendido=comando_encendido, timeout=STD_TIMEOUT,
+            )
+            tel_vex04P04 = await self.assert_next_sample(topic=self.remote.tel_vex04P04, flush=True,)
+            tel_vex04P04 = await self.assert_next_sample(topic=self.remote.tel_vex04P04, flush=True,)
+            self.assertTrue(tel_vex04P04.fallaTermica is False)
+            self.assertTrue(tel_vex04P04.comandoEncendido == comando_encendido)
+
+    async def test_do_vin01P01(self):
+        async with self.make_csc(initial_state=salobj.State.STANDBY, config_dir=None, simulation_mode=1):
+            await salobj.set_summary_state(remote=self.remote, state=salobj.State.ENABLED)
+
+            tel_vin01P01 = await self.assert_next_sample(topic=self.remote.tel_vin01P01, flush=True,)
+            self.assertTrue(tel_vin01P01.estadoFuncionamiento is False)
+            self.assertTrue(tel_vin01P01.estadoSelector is False)
+            self.assertTrue(tel_vin01P01.comandoEncendido is False)
+
+            comando_encendido = True
+            await self.remote.cmd_vin01P01.set_start(
+                comandoEncendido=comando_encendido, timeout=STD_TIMEOUT,
+            )
+            tel_vin01P01 = await self.assert_next_sample(topic=self.remote.tel_vin01P01, flush=True,)
+            tel_vin01P01 = await self.assert_next_sample(topic=self.remote.tel_vin01P01, flush=True,)
+            self.assertTrue(tel_vin01P01.estadoFuncionamiento == comando_encendido)
+            self.assertTrue(tel_vin01P01.estadoSelector is False)
+            self.assertTrue(tel_vin01P01.comandoEncendido == comando_encendido)
```

Comment:
I see your point about repetition and I agree it's probably easy to automate these tests cases by putting all the logic into one `check_command` that takes the topic name and a set of new values.

You will have to be a bit careful to handle floats with "assertAlmostEqual" but all other values can be fed to `assert_next_sample` as a keyword dict.

Reply:
See my previous comments. I made better use of `assert_next_sample`. I may consider automating the test cases but I prefer to do that in a separate ticket.